lib/cellect/server/workflow.rb

```diff
@@ -41,23 +46,39 @@ def initialize(name, pairwise: false, prioritized: false)
         self.users = { }
         self.pairwise = !!pairwise
         self.prioritized = !!prioritized
-        self.subjects = set_klass.new
+        self.subjects ||= set_klass.new
+        self.state = :initializing
+        self.can_reload = false
       end
 
       # Loads subjects from the adapter
       def load_data
-        self.state = :initializing
-        self.subjects = set_klass.new
-        Cellect::Server.adapter.load_data_for(name).each do |hash|
-          subjects.add hash['id'], hash['priority']
-        end
+        return if self.state == :ready
+        self.state = :loading
+        load_adapter_data(self.subjects)
+        reset_can_reload_timer
         self.state = :ready
       end
 
+      # Reloads subjects from the adapter
+      def reload_data
+        if can_reload_data?
+          self.can_reload = false
+          self.state = :reloading
+          new_data = self.subjects.class.new
```

Comment:
I'm totally unclear how this works, and how it makes it so that it runs async and not blocking lookups on this actor. Is it doing that?